### PR TITLE
Initial commit of clamav

### PR DESCRIPTION
**Pull request by dasJ**

### CODE REVIEW

File: meta/main.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+galaxy_info:
+  author: Janne Heß
+  description: Install and configure a clamav daemon with freshclam
+  company: stuvus (http://stuvus.uni-stuttgart.de)
```
Comment (SlothOfAnarchy) [inline meta/main.yml]: Are you not using our latest ansible skeleton?

File: tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  template:
+    src: tmpfiles.conf.j2
+    dest: /etc/tmpfiles.d/clamav.conf
+    owner: root
+    group: root
+    mode: 0644
+  register: tmpfiles
+
+- name: Create tmpfiles
+  command: systemd-tmpfiles --create
```
Comment (SlothOfAnarchy) [inline tasks/main.yml:58]: Did you try the `tempfile` module?

File: tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+- name: Reload systemd
+  command: systemctl daemon-reload
+  when: clamd_unit.changed
+
+- name: Download initial clam databases
+  command:
+    cmd: freshclamd
+    creates: /var/lib/clamav/main.cvd
+
+- name: Start services
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: Is this necessary when you already have a handler for that?

File: tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    group: root
+    mode: 0644
+  register: tmpfiles
+
+- name: Create tmpfiles
+  command: systemd-tmpfiles --create
+  when: tmpfiles.changed
+
+- name: Reload systemd
+  command: systemctl daemon-reload
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: Ansible also has a module to reload daemons.

File: templates/clamd.conf.j2 [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# {{ ansible_managed }}
+
+LocalSocket              /run/clamav/clamd.sock
+PidFile                  /run/clamav/clamd.pid
+TemporaryDirectory       /tmp
+LogFile                  /var/log/clamav/clamd.log
+LogSyslog                true
+User                     clamav
```
Comment (SlothOfAnarchy) [inline templates/clamd.conf.j2:8]: Is the user created automatically?
Also, these values should be configurable.

**Review by SlothOfAnarchy — COMMENTED** [review]
nice and clean nonetheless

File: tasks/main.yml (hunk 4 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: systemd:
    state: restarted
    daemon_reload: yes

File: tasks/main.yml (hunk 2 above)
Comment (Scriptkiddi) [reply tasks/main.yml:58]: he is creating systemd-tmpfile not tmp files

File: meta/main.yml (hunk 1 above)
Comment (dasJ) [reply meta/main.yml]: I obviously don't

File: tasks/main.yml (hunk 2 above)
Comment (dasJ) [reply tasks/main.yml:58]: Yup, both are for completely different purposes. The don't have anything in common except for the similar name

File: tasks/main.yml (hunk 4 above)
Comment (dasJ) [reply tasks/main.yml]: Nice to know

File: tasks/main.yml (hunk 3 above)
Comment (dasJ) [reply tasks/main.yml]: Yes

File: templates/clamd.conf.j2 (hunk 5 above)
Comment (dasJ) [reply templates/clamd.conf.j2:8]: Please don't make me to. It's one of the few things **most** distributions agree about.

Comment (dasJ) [reply templates/clamd.conf.j2:8]: They even agree about the name of the user (I think even Debian which is the distribution where **everything** has a different name)

File: tasks/main.yml (hunk 4 above)
Comment (SlothOfAnarchy) [reply tasks/main.yml]: Is there a problem with it? Otherwise please use it.

File: tasks/main.yml (hunk 3 above)
Comment (SlothOfAnarchy) [reply tasks/main.yml]: How so?

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

File: tasks/main.yml (hunk 3 above)
Comment (dasJ) [reply tasks/main.yml]: The config may not be changed but clam may not be running

**Comment by dasJ** [comment]
@SlothOfAnarchy Pls rereview

File: tasks/main.yml (hunk 3 above)
Comment (SlothOfAnarchy) [reply tasks/main.yml]: If you add `enabled: true` to the handler we don`t need the start task.
When the config is copied the first time, it most definitely is changed.

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]
See [my previous comment](https://github.com/stuvusIT/clamav/pull/1/files#r138404776)

File: tasks/main.yml (hunk 3 above)
Comment (dasJ) [reply tasks/main.yml]: But then the handler is changed when the unit was disabled and not running.

Comment (SlothOfAnarchy) [reply tasks/main.yml]: Your task would also be changed?

Comment (dasJ) [reply tasks/main.yml]: Okay. Proper use case: No configuration was changed, but clamd was not running, probably because it failed

Comment (dasJ) [reply tasks/main.yml]: Okay. Proper use case: No configuration was changed, but clamd was not running, probably because it failed

Comment (SlothOfAnarchy) [reply tasks/main.yml]: Is clamd prone to fail?
Wouldn`t proper monitoring or general automatic restarting of services be a better solution? :P

Comment (dasJ) [reply tasks/main.yml]: Bruh.

Comment (dasJ) [reply tasks/main.yml]: Maybe a dumb admin stopped it because it was causing too much CPU utilitation

Comment (SlothOfAnarchy) [reply tasks/main.yml]: I don't think we will ever finish this project if we have to account for dumb admins in every role.

Comment (dasJ) [reply tasks/main.yml]: We'll not finish it either if you don't merge this

**Review by SlothOfAnarchy — APPROVED** [review]

**Pull request merged**